### PR TITLE
test: create e2e tests for package, fix found errors

### PR DESCRIPTION
**Pull request by kuskoman**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,8 @@
         "lint:diagnostics": "ttsc -p tsconfig.diagnostics.json",
         "test": "jest --passWithNoTests",
         "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+        "test:e2e": "jest --forceExit --config ./jest.config.e2e.ts",
```
Comment (kopecmaciej) [inline package.json:22]: a tu ten --forceExit to dobry pomysł? może lepiej wiedzieć, że jakiś resource jest niezamkniety

**Comment by ssukienn** [comment]
I think the contract on how to use log functions with extra parameters is unclear

**Comment by kuskoman** [comment]
@ssukienn I also think so, but Nest decided to use it like that :P

File: src/logger.ts [hunk 2]
```diff
@@ -133,53 +137,104 @@ export class NestTransportLogger implements LoggerService {
...
+                    baseContext: args[0],
+                };
+            }
+
+            if (args.length === 2) {
+                return {
+                    messages: [errMessage],
+                    stacktrace,
+                    baseContext: args[1],
+                    additionalContext: args[0],
```
[the remark this answers is not in the document]
Comment (kuskoman) [reply src/logger.ts:169]: @ssukienn nest puts base context as the last argument of log

**Comment by kuskoman** [comment]
btw. the contract is defined here https://github.com/nestjs/nest/blob/master/packages/common/services/logger.service.ts i cannot change it

**Review by kopecmaciej — APPROVED** [review]

**Pull request merged**